### DISCUSSION TITLE
Support of LSP Semantic Tokens Customization?

### DISCUSSION THREAD

**Discussion by chikko80** [NvChad/NvChad#1872] [General]
I recently switched from VsCode and first started with my own Nvim Config.
Yesterday i tried NvChad and really liked it so i thought i'll give it a go.

My first nvim config i set up with this theme:

[https://github.com/olimorris/onedarkpro.nvim
](https://github.com/olimorris/onedarkpro.nvim
)

since this is the theme I used for years in vscode.

Like VsCode the theme supports semantic token customization where you can override specific lsp semantic tokens.
I've been using personal syntax coloring for years so i was wondering if this is possible with NvChad currently?

I tried to set the config in highlights.lua:

```lua
---@type Base46HLGroupsList
M.override = {
    Comment = {
        italic = true,
    },

    ["@lsp.type.method"] = { fg = "#61AFEF" },
    ["@lsp.typemod.method"] = { fg = "#61AFEF" },

    ["@lsp.type.variable"] = { fg = "#B1B8C4" },
    ["@lsp.type.generic"] = { fg = "#B1B8C4" },
    ["@lsp.type.parameter"] = { fg = "#D19A66" },

    ["@lsp.type.selfKeyword"] = { fg = "#C678DD" },
    ["@lsp.type.keyword"] = { fg = "#C678DD" },
    ["@lsp.type.selfTypeKeyword"] = { fg = "#C678DD" },
    ["@lsp.type.formatSpecifier"] = { fg = "#C678DD" },

    ["@lsp.type.struct"] = { fg = "#E5C07B" },
    ["@lsp.type.typeAlias"] = { fg = "#E5C07B" },
    ["@lsp.type.namespace"] = { fg = "#E5C07B" },
    ["@lsp.type.library"] = { fg = "#E5C07B" },
    ["@lsp.typemod.namespace"] = { fg = "#E5C07B" },
    ["@lsp.type.property"] = { fg = "#D5B3FF" },
    ["@lsp.type.interface"] = { fg = "#2ACFAB" },

    ["@lsp.type.enumMember"] = { fg = "#56B6C2" },

    ["@lsp.type.macro"] = { fg = "#D19A66" },
    ["@lsp.mod.constant"] = { fg = "#D19A66" },
    ["@lsp.typemod.variable.constant"] = { fg = "#D19A66" },
}
```

but it doesn't work very well. Some colors get applied, some don't.

**Comment by siduck** [comment]
@chikko80 I already do, use neovim 0.9

look https://github.com/NvChad/base46/blob/v2.0/lua/base46/integrations/syntax.lua#L120

**Reply by chikko80** [reply]
@siduck 

### Edit: It was the treesitter default config which caused the issue, i messed up my override

Is there anything else that could conflict?

<img width="522" alt="Screenshot 2023-03-30 at 3 05 17 PM" src="https://user-images.githubusercontent.com/58805871/228845038-069c0f19-957a-4ff4-8c41-dbeb21b67f77.png">

I am clearly overriding the enumMember token in highlights/lua (M.override{})

<img width="802" alt="Screenshot 2023-03-30 at 3 06 51 PM" src="https://user-images.githubusercontent.com/58805871/228845455-88bff923-402c-49c5-961b-368ee370b370.png">

the the enumMember is still red

**Reply by siduck** [reply]
@chikko80 make sure those exist on our repo. and enumMember.rust doesnt btw, so it must be put into hl_add field.

**Reply by chikko80** [reply]
Got it, that was my next question, but you already answered it thanks. Now everything is clear
